src/main/java/frc/trigon/robot/subsystems/roller/simulationroller/SimulationRollerIO.java

```diff
@@ -0,0 +1,83 @@
+package frc.trigon.robot.subsystems.roller.simulationroller;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.simulation.DCMotorSim;
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+import frc.trigon.robot.constants.RobotConstants;
+import frc.trigon.robot.subsystems.roller.RollerIO;
+import frc.trigon.robot.subsystems.roller.RollerInputsAutoLogged;
+import frc.trigon.robot.utilities.Conversions;
+import org.littletonrobotics.junction.Logger;
+
+public class SimulationRollerIO extends RollerIO {
+    private final SingleJointedArmSim angleSimulation = SimulationRollerConstants.ANGLE_MOTOR;
+    private final DCMotorSim collectorSimulation = SimulationRollerConstants.COLLECTOR_MOTOR;
+
+    private double angleMotorVoltage, collectorMotorVoltage;
+
+    @Override
+    protected void updateInputs(RollerInputsAutoLogged inputs) {
+        angleSimulation.update(RobotConstants.PERIODIC_TIME_SECONDS);
+        collectorSimulation.update(RobotConstants.PERIODIC_TIME_SECONDS);
+
+        inputs.angleMotorVoltage = angleMotorVoltage;
```

Comment:
here, check if the robot is disabled (DriverStation.isDisabled()). If it is, stop the motors. You need to do this since in the real robot this will happen, but not the sim.